### ISSUE TITLE
Light/Status, Critical doesn't meet AA requirements

### ISSUE THREAD

**Issue by lorbritton** [hpe-design/design-system#945]
Tested the Light/Status, Critical swatch in Figma to make sure it meets AA color contrast requirements, and it came back as Failed at 2.99/1 instead of 3/1.

**Comment by gregfuruiye** [comment]
Resources:
[https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html)
[https://www.w3.org/WAI/WCAG21/Techniques/general/G183](https://www.w3.org/WAI/WCAG21/Techniques/general/G183)
[https://webaim.org/articles/contrast/#sc143](https://webaim.org/articles/contrast/#sc143)

One consideration for all of the status colors are what they are to be used for. In our case, status colors are not to be used for text. Rather, they are to be used for background or with icons that have associated information available (i.e. a description of the condition in text form). In these cases, the colors currently meet the WCAG 2.1 conditions when applied appropriately.

For example, all of the black text over the status color meets the guidelines and should be used as such ('text' or 'strong'). However, when using a status color for an icon, additional information must be included since the status colored icon alone does not meet the criteria. The image below is an example of how we are meeting compliance in light mode.

<img width="319" alt="status" src="https://user-images.githubusercontent.com/17889451/87601710-dcda6500-c6b2-11ea-96c3-bc9f03c4ab45.png">

**Comment by MikeWalrath** [comment]
Thanks Greg. This information could also be leveraged/referenced as we create the Accessibility and Notification pages pages in the future. 

@lorbritton any questions on the above?

**Comment by lorbritton** [comment]
Another consideration is that people will inevitably use design system components in ways that weren't originally intended. With things like color contrast it's best to bulletproof it as much as possible, especially since it impacts if people with low vision can see something or not. It's a small change for us to make the colors meet the 3:1 criteria when using white or black, but it can potentially make a big impact for our users.

**Comment by lorbritton** [comment]
For example, if the color is changed to #D93B3B, then it would pass AA compliance for use with both white and black text/backgrounds at all sizes. <img width="524" alt="Screen Shot 2020-07-15 at 4 31 40 PM" src="https://user-images.githubusercontent.com/51830207/87606666-c931fd00-c6b8-11ea-982e-88f1c256d158.png">

**Comment by gregfuruiye** [comment]
I agree that it is a small change, but only for the critical color. Warning and OK status would need to see significant changes for white text to occur. Like you suggested, people will use the design system components incorrectly. In the case for button color we saw on the double confirmation, we agreed to use a red color, but did not specify the use of white text. We also allow for 'color' button within the design system. To what extent do we move to satisfy all potential use cases.

**Comment by lorbritton** [comment]
I would aim for radically inclusive like Microsoft's Design System, Fluent–it goes back to that inclusive design principle of designing for the marginalized 20% instead of the 80%. Microsoft's color palette is essentially grayscale plus a theme color (blue for them, green for us) and accessible status colors if you look at their Figma file. I downloaded their Figma file [**here**](https://www.figma.com/file/AIOrcOmgi2aPORsCymbwv7/Microsoft-Fluent-Web?node-id=3252%3A0) if you want to check it out.

**Comment by gregfuruiye** [comment]
Just to be clear, I think the color contrast you are quoting in the image is for white text (#ffffff), correct? Black contrast ratio for the current critical status color is 7.01:1

**Comment by gregfuruiye** [comment]
Yeah, I have that Figma file as well from a ways back. I'll go back and review it again. I agree with you, however, could it also be argued that these colors could be misused as well and miss meeting the standard.

**Comment by lorbritton** [comment]
Correct, 2.99:1 for white text and 7.02:1 for black text. For the proposed color the contrast ratio is 4.63:1 for black text and 4.52:1 for white text.

**Comment by gregfuruiye** [comment]
The color palette helps identify the use of the color which is nice. "Status Severe Warning Type or Icon"

**Comment by gregfuruiye** [comment]
I've tweaked out the critical status color to meet AA guidelines when using text strong for both dark and light modes. This allows for the use of white text over the critical status color background when using the critical status color for a button color. It should be noted that we need to provide clear guidance on the use of status colors (i.e. do not use for text).

Light mode critical status color = FC5A5A (light/text strong = 6.75:1, dark/text strong = 3.1:1)
Dark mode critical status color = D04F4E (dark/background = 3.11:1, dark/text strong = 4.28:1)

Using dark/background for text color was to align color to the dark mode background. light/text strong = 4.91:1 but the use of black on the critical color with the background color doesn't align well.

[https://www.figma.com/file/eZYR3dtWdb9U90QvJ7p3T9/HPE-Color-Styles?node-id=1416%3A402](https://www.figma.com/file/eZYR3dtWdb9U90QvJ7p3T9/HPE-Color-Styles?node-id=1416%3A402)

**Comment by MikeWalrath** [comment]
Thanks Greg. Did you happen to open an issue to update the theme? Where do you think would be the best place to document the use of the status colors? 

I think it would be great to test out the new MDX update and add copy to the [Color](https://design-system.hpe.design/foundation/color#status-colors) page?

I think it would also be good to add it to the Figma file and Notifications template when we work on it. Just to try and over communicate the correct use.

**Comment by gregfuruiye** [comment]
I have not created an issue yet, I was proposing these colors for review. If there's no real feedback after today, then I'll move forward and update the Figma file and create an issue to update the theme.
